docs/MIGRATING.md

```diff
@@ -0,0 +1,5 @@
+# Migrating docs for callers of the wasmvm Go project
```

Comment:
Nice 👍
We'll have to add a few more entries here (in a separate PR). For example the gas multiplier reduction

Reply:
Yeah, wanted to take the opportunity to create a place for those things.